forc-pkg/src/pkg.rs

```diff
@@ -2369,14 +2402,66 @@ pub fn build_with_options(build_options: BuildOpts) -> Result<Built> {
     };
 
     let manifest_file = ManifestFile::from_dir(&this_dir)?;
+    let member_manifests = manifest_file.member_manifests()?;
+    let lock_path = manifest_file.lock_path()?;
+    let build_plan = BuildPlan::from_lock_and_manifests(
+        &lock_path,
+        &member_manifests,
+        build_options.pkg.locked,
+        build_options.pkg.offline,
+    )?;
+    let graph = build_plan.graph();
+    let manifest_map = build_plan.manifest_map();
+    let build_profiles: HashMap<String, BuildProfile> = build_plan.build_profiles().collect();
+    // Get the selected build profile using build options
+    let (profile_name, build_profile) = build_profile_from_opts(&build_profiles, &build_options)?;
+
+    // If this is a workspace we want to have all members in the output.
+    let outputs = match &manifest_file {
+        ManifestFile::Package(pkg_manifest) => std::iter::once(
+            build_plan
+                .find_member_index(&pkg_manifest.project.name)
+                .ok_or_else(|| anyhow!("Cannot found project node in the graph"))?,
+        )
+        .collect(),
+        ManifestFile::Workspace(_) => build_plan.member_nodes().collect(),
+    };
+    // Build it!
+    let mut built_workspaces = HashMap::new();
+    let built_packages_with_source_map = build(&build_plan, &build_profile, &outputs)?;
+    let output_dir = build_options
+        .clone()
+        .pkg
+        .output_directory
+        .map(PathBuf::from);
+    for (node_ix, (built_package, source_map)) in built_packages_with_source_map
+        .iter()
```

Comment:
We should use `into_iter` here to take the elements by value and avoid the need to clone the entirety of each package below.
```suggestion
        .into_iter()
```